tests/compilation_cache_test.py

```diff
@@ -179,6 +181,22 @@ def test_put_executable(self):
           actual = jax.lib.xla_client.execute_with_python_values(deserialized_executable, inputs_to_executable, backend)
           self.assertEqual(expected, actual)
 
+  def test_pmap(self):
+      cc._cache = None
+      with tempfile.TemporaryDirectory() as tmpdir:
+          cc.initialize_cache(tmpdir)
+          f = pmap(lambda x: x - lax.psum(x, 'i'), axis_name='i')
+          shape = (jax.lib.xla_bridge.device_count(), 4)
+          x = np.arange(prod(shape), dtype=np.float32).reshape(shape)
+          f(x)
+          files_in_directory = len(os.listdir(tmpdir))
+          self.assertEqual(files_in_directory, 1)
+          x = np.arange(prod(shape), dtype=np.int32).reshape(shape)
+          f(x)
+          files_in_directory = len(os.listdir(tmpdir))
+          self.assertEqual(files_in_directory, 2)
+          #TODO: create a test for calling pmap more than once
```

Comment:
nit: say "for calling pmap with the same input more than once", and specify that we can't test this now due to the in-memory cache.